common/djangoapps/util/request_rate_limiter.py

```diff
@@ -101,3 +103,18 @@ def tick_request_counter(self, request):
         """
         for key in self.keys_to_check(request):
             self.cache_incr(key)
+
+
+class LoginAndRegisterRateLimiter(RequestRateLimiter):
```

Comment:
These rate limiters that are build on top of the `django-ratelimit-backend` library are meant to be used as a part of the authentication flow.  In this case we want to just rate-limit a single view based on ip.  I think the `django-ratelimit` library is better suited for this.  Is there a reason you're not using that library, it seems like a decorator around the view from that library should be all we need here.

Django Ratelimit docs: https://django-ratelimit.readthedocs.io/en/stable/

Reply:
@feanil No specific reason, `django-ratelimit-backend` is mostly used in platform to ratelimit the views and that's the only reason I have chosen this over `django-ratelimit`.